core/contracts/oracle/implementation/Voting.sol

```diff
@@ -388,15 +394,15 @@ contract Voting is Testable, Ownable, OracleInterface, VotingInterface {
             keccak256(abi.encodePacked(price, salt, msg.sender, time, roundId, identifier)) == voteSubmission.commit,
             "Revealed data != commit hash"
         );
-        delete voteSubmission.commit;
 
-        // Lock in round variables including snapshotId and inflation rate. Not that this will only execute a snapshot
-        // if the `snapshotCurrentRound` function was not already called for this round.
-        _freezeRoundVariables(roundId);
+        // To protect against flash loans, we require snapshot be validated as EOA.
+        require(rounds[roundId].snapshotId != 0, "Round must be snapshotted before reveal can happen");
```

Comment:
This require message is too long. All require messages must be _at most_ 32 chars or they take up extra byte code in the contracts. All messages <=32 chars use the same storage size.
```suggestion
        require(rounds[roundId].snapshotId != 0, "Round has no snapshot");
```